flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/RecordWriterOutput.java

```diff
@@ -72,6 +76,11 @@ public RecordWriterOutput(
 
```

Comment:
I can remove the reduplication on `RecordWriterOutput`, `ChainingOutput` and `CopyingChainingOutput`. Basically those `Outputs` that don't forward to other outputs but instead push into the operator or into the network. For the other `Outputs` removing the duplication is not possible because inside the respective `output()` method they call `output()` of another `Output`. They call either with an `OutputTag` or without, so the method body is not actually a duplicate.

I did find another bug, though, where `CopyingBroadcastingOutputCollector` in `OperatorChain` was not calling the correct `collect()` method on the downstream `Outputs`. 😃